web_diagram_position/static/src/js/diagram_controller.js

```diff
@@ -0,0 +1,208 @@
+// Copyright 2017 - 2018 Modoolar <info@modoolar.com>
+// License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html).
+odoo.define('web_diagram_position.DiagramController', function (require) {
+"use strict";
+
+var core = require('web.core');
+var Dialog = require('web.Dialog');
+var dialogs = require('web.view_dialogs');
+
+var _t = core._t;
+
+require('web_diagram.DiagramController').include({
+
+    custom_events: {
+        add_edge: '_onAddEdge',
+        edit_edge: '_onEditEdge',
+        edit_node: '_onEditNode',
+        remove_edge: '_onRemoveEdge',
+        remove_node: '_onRemoveNode',
+    },
+
+    is_readonly: function () {
+        var context = this.context;
+        return context ? context.edit != undefined && !context.edit : false;
+    },
+
+    _addNode: function () {
+        if (this.model.modelName != 'project.workflow')
+            return this._super();
```

Comment:
I am not sure but may b you can remove this lines. Can you please elaborate this changes, According to above lines your changes will only affect for project.workflow. Correct me if i am wrong here.